src/ptx.jl

```diff
@@ -86,6 +89,29 @@ runtime_slug(@nospecialize(job::CompilerJob{PTXCompilerTarget})) =
        "-debuginfo=$(Int(llvm_debug_info(job)))" *
        "-exitable=$(job.target.exitable)"
 
+# We have two code caches so that functions can be forced inline even if they
+# have been compiled before.
+const PTX_CI_CACHE = CodeCache()
+const PTX_ALWAYS_INLINE_CI_CACHE = CodeCache()
```

Comment:
This doesn't seems scalable...

@aviatesk Is it expected that optimization params can leak into the inference cache?

Reply:
Yes I think that's expected. Currently the cache assumes that the same optimization params are used through-out.

Reply:
probably yes, that would also decouple it from the alwaysinline setting specifically

Reply:
I think so, yeah. The problem is now that CUDA.jl (and other back-ends) override this function: https://github.com/JuliaGPU/CUDA.jl/blob/bd87c0e97cf29788fb68752f52cacfcaea3fadb6/src/compiler/gpucompiler.jl#L75=. Maybe it would be better to include the type of the compiler target so that we don't have to override this function at all?

Reply:
I like that idea.  I have updated the pr to include the type of the compiler target in the hash for the cache.